src/client/views/components/todoItem.jsx

```diff
@@ -75,12 +75,16 @@ export default React.createClass({
     },
 
     handleBack: function () {
-        const {description, dueDate, title} = this.props;
-        this.setState({
-            inEdit: false,
-            description,
-            dueDate: dueDate.toDateString(),
-            title});
+        const {description, dueDate, title, addingNewItem} = this.props;
+        if (addingNewItem) {
+            this.handleDelete();
+        } else {
+            this.setState({
+                inEdit: false,
+                description,
+                dueDate: dueDate.toDateString(),
+                title});
+        }
```

Comment:
I think that when skimming through the code it's easier to miss the `return` statement and misinterpret the code, while the `else` here makes the intent more clear. I'm inclined to leave this as it is.